src/main/kotlin/tanvd/grazi/ide/ui/rules/GraziRulesTree.kt

```diff
@@ -125,7 +126,7 @@ class GraziRulesTree(selectionListener: (meta: Any) -> Unit) : Disposable {
         }
 
     val isModified: Boolean
-        get() = state.isNotEmpty()
+        get() = state.values.any { it.lang in langs }
```

Comment:
If we filter for everywhere state to provide us information only on langs in `langs` field -- probably we may encapsulate it into the getter of `state`? On each get it will return filtered HashSet only with currently unabled languages

Reply:
Unfortunately we can't encapsulate it in state, cause we need to add and remove values to it. Also not everywhere, only there and in state.